### PR TITLE
update beaconStateHandler to make use of DataProvider abstraction

### PR DESCRIPTION
**Pull request by rolfyone**
Updated BeaconStateHandler and created the BeaconState type in DataProvider schema.

Refactored BeaconValidators, BeaconBlockRoot because they both used internal BeaconState.

addresses #1315

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BeaconValidators.java [hunk 1]
```diff
@@ -105,7 +114,12 @@ public static long getEffectiveListSize(
     if (!activeOnly) {
       return list.size();
     } else {
-      return list.stream().filter(v -> ValidatorsUtil.is_active_validator(v, epoch)).count();
+      return list.stream().filter(v -> is_active_validator(v, epoch)).count();
     }
   }
+
+  private static boolean is_active_validator(Validator validator, UnsignedLong epoch) {
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/BeaconValidators.java:121]: we don't want to use ValidatorsUtil?

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/Fork.java [hunk 2]
```diff
@@ -11,18 +11,19 @@
...
 
-import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.databind.JsonSerializer;
-import com.fasterxml.jackson.databind.SerializerProvider;
-import java.io.IOException;
-import tech.pegasys.artemis.util.bls.BLSPublicKey;
+import com.google.common.primitives.UnsignedLong;
+import tech.pegasys.artemis.util.SSZTypes.Bytes4;
 
-public class BLSPublicKeySerializer extends JsonSerializer<BLSPublicKey> {
```
Comment (macfarla) [inline data/provider/src/main/java/tech/pegasys/artemis/api/schema/Fork.java:22]: BLSPublicKeySerializer renamed as Fork is confusing

File: data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandler.java [hunk 3]
```diff
@@ -81,48 +82,40 @@ public BeaconStateHandler(
...
       }
-      if (!combinedClient.isStoreAvailable()) {
+      if (!provider.isStoreAvailable()) {
         ctx.status(SC_NO_CONTENT);
         return;
       }
 
       if (parameters.containsKey(ROOT)) {
-        future = queryByRootHash(validateQueryParameter(parameters, ROOT));
+        future = provider.getStateByBlockRoot(getParameterValueAsBytes32(parameters, ROOT));
```
Comment (macfarla) [inline data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandler.java:95]: nice

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java [hunk 4]
```diff
@@ -87,86 +91,67 @@ public void shouldReturnNotFoundWhenQueryAgainstMissingRootObject() throws Excep
...
-        new BeaconStateHandler(combinedChainDataClient, jsonProvider);
+    final BeaconStateHandler handler = new BeaconStateHandler(provider, jsonProvider);
     when(context.queryParamMap()).thenReturn(Map.of(ROOT, List.of(blockRoot.toHexString())));
 
     handler.handle(context);
 
     verify(context).result(args.capture());
     SafeFuture<String> data = args.getValue();
-    assertEquals(data.get(), jsonProvider.objectToJSON(beaconState));
+    assertEquals(data.get(), jsonProvider.objectToJSON(new BeaconState(beaconStateInternal)));
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java]: do we need to create a new BeaconState here?

Comment (macfarla) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java]: nvm I get it. beaconStateInternal is the internal data structure

Comment (macfarla) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java]: might make it clearer to make it a method call getMeTheExternalRepresentationOfTheInternalState(beaconStateInternal)

**Review by macfarla — APPROVED** [review]
looks good overall. few comments

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/BeaconValidators.java (hunk 1 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/artemis/api/schema/BeaconValidators.java:121]: its a different class...

File: data/provider/src/main/java/tech/pegasys/artemis/api/schema/Fork.java (hunk 2 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/artemis/api/schema/Fork.java:22]: wow, that's not what happened at all, one got deleted and one got added... not sure why it'd think it's renamed...

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java (hunk 4 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java]: we're constructing the actual object we can use... i think the bigger problem is where we mix the handling of both objects, then it gets confusing... i'm trying to clean it up, but its not quite right yet...

Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java]: I just realised that object is already constructed above, i'll clean this one up and it'll be more clear...

**Pull request merged**